stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/transactions.py

```diff
@@ -30,115 +32,191 @@ class TransactionsClient(BaseTransactionsClient):
+                "match_mapping_type": "string",
+                "match": "description",
+                "mapping": {"type": "text"},
             }
-        }
-
-        _ = self.client.indices.create(
+        },
+        {
+            "titles": {
+                "match_mapping_type": "string",
+                "match": "title",
+                "mapping": {"type": "text"},
+            }
+        },
+        # Projection Extension https://github.com/stac-extensions/projection
+        {"proj_epsg": {"match": "proj:epsg", "mapping": {"type": "integer"}}},
+        {
+            "proj_projjson": {
+                "match": "proj:projjson",
+                "mapping": {"type": "object", "enabled": False},
+            }
+        },
+        {
+            "proj_centroid": {
+                "match_mapping_type": "string",
+                "match": "proj:centroid",
+                "mapping": {"type": "geo_point"},
+            }
+        },
+        {
+            "proj_geometry": {
+                "match_mapping_type": "string",
+                "match": "proj:geometry",
+                "mapping": {"type": "geo_shape"},
+            }
+        },
+        {
+            "no_index_href": {
+                "match": "href",
+                "mapping": {"type": "text", "index": False},
+            }
+        },
+        # Default all other strings not otherwise specified to keyword
+        {"strings": {"match_mapping_type": "string", "mapping": {"type": "keyword"}}},
+        {"numerics": {"match_mapping_type": "long", "mapping": {"type": "float"}}},
+    ]
+
+    mappings = {
+        "numeric_detection": False,
+        "dynamic_templates": dynamicTemplates,
+        "properties": {
+            "geometry": {"type": "geo_shape"},
+            "assets": {"type": "object", "enabled": False},
+            "links": {"type": "object", "enabled": False},
+            "properties": {
+                "type": "object",
+                "properties": {
+                    # Common https://github.com/radiantearth/stac-spec/blob/master/item-spec/common-metadata.md
+                    "datetime": {"type": "date"},
+                    "start_datetime": {"type": "date"},
+                    "end_datetime": {"type": "date"},
+                    "created": {"type": "date"},
+                    "updated": {"type": "date"},
+                    # Satellite Extension https://github.com/stac-extensions/sat
+                    "sat:absolute_orbit": {"type": "integer"},
+                    "sat:relative_orbit": {"type": "integer"},
+                },
+            },
+        },
+    }
+
+    def create_item_index(self):
+        """Create the index for Items."""
+        self.client.indices.create(
             index="stac_items",
-            body=mapping,
+            body={"mappings": self.mappings},
             ignore=400,  # ignore 400 already exists code
         )
 
-    def create_item(self, model: stac_types.Item, **kwargs):
+    @overrides
```

Comment:
renamed the parameters for all of these methods that override ones in the base class to match the names in the base class, and put them in the same order they are in the base class.